importer/trickle/trickledag.go

```diff
@@ -65,25 +54,35 @@ func Layout(db *h.DagBuilderHelper) (ipld.Node, error) {
 	return out, nil
 }
 
-func fillTrickleRec(db *h.DagBuilderHelper, node *h.UnixfsNode, depth int) error {
+// fillTrickleRec creates a trickle (sub-)tree with an optional maximum specified depth
+// in the case maxDepth is greater than zero, or with unlimited depth otherwise
+// (where the DAG builder will signal the end of data to end the function).
+func fillTrickleRec(db *h.DagBuilderHelper, node *h.UnixfsNode, maxDepth int) error {
 	// Always do this, even in the base case
 	if err := db.FillNodeLayer(node); err != nil {
 		return err
 	}
 
-	for i := 1; i < depth && !db.Done(); i++ {
-		for j := 0; j < layerRepeat && !db.Done(); j++ {
-			next := db.NewUnixfsNode()
-			if err := fillTrickleRec(db, next, i); err != nil {
+	for depth := 1; ; depth++ {
+		// Apply depth limit only if the parameter is set (> 0).
+		if maxDepth > 0 && depth == maxDepth {
+			return nil
+		}
+		for layer := 0; layer < layerRepeat; layer++ {
+			if db.Done() {
+				return nil
+			}
```

Comment:
Why move do this separately?

Reply:
To avoid repeating the same condition in both `for`s, and make it more clear that if `db.Done` is true nothing more needs to be done in this function and it should just exit there (removing the final `return` from the code).